zowe/api-layer
# Resolve vulnerabilities in npm packages

## Problem 
There were security vulnerabilities in our npm package dependencies used for UI. One of them was also reported by the GitHub security alert, related to the `lodash` version used which contained an high severity vulnerability. 

## Solution
Different npm dependencies have been updated to the latest version in order to solve this problem.

**Before** updating them, there were:

138 **Low severity** vulnerabilities
2 **Moderate severity** vulnerabilities
5 **High severity** vulnerabilities

**After** updating them, there are just 3 vulnerabilities:
1 **High severity** (coming from cypress, and according to [https://github.com/cypress-io/cypress/issues/4730](https://github.com/cypress-io/cypress/issues/4730) will be fixed soon)
1 **Low severity** 
1 **Moderate severity**


The `lodash` package has been also updated to a newest version.
The moderate one should be fixed in the future after the try-out branch is merged, since it comes from the `swagger-ui` library which will be replaced. 


## How to test

1. Delete the `node_modules` folder from the `frontend` directory
2. Delete the `package-lock.json` file (not the `package.json`)
3. Run `npm install`
4. Run `npm audit` from the `api-catalog-ui/frontend` folder to check the security vulnerabilities.

The test has been done using Nodejs v. 8.11.1, the same used on mainframe.
